src/main/java/org/schemaspy/service/DatabaseService.java

```diff
@@ -84,42 +84,31 @@ public void gatheringSchemaDetails(Config config, Database db, ProgressListener
    private void initCatalogs(Config config, Database db, ProgressListener listener) throws SQLException {
 
             String sql = Config.getInstance().getDbProperties().getProperty("selectCatalogsSql");
-            PreparedStatement stmt = null;
-			ResultSet rs =  null;
+
             if (sql != null && db.getCatalog() != null) {
-                try {
-                stmt = sqlService.prepareStatement(sql,db,null);
-                rs = stmt.executeQuery();
-                    while (rs.next()) {
+                try (PreparedStatement stmt = sqlService.prepareStatement(sql, db, null);
+                     ResultSet rs = stmt.executeQuery();) {
+                    if (rs.next()) {
                          db.getCatalog().setComment(rs.getString("catalog_comment"));
-                         break;
                     }
                 } catch (SQLException sqlException) {
                     //db.getSchema().setComment(null);
```

Comment:
@danwatt instead leave this `catch `empty as it was could you change the code to something like:

```
String msg = listener.recoverableExceptionEncountered("Failed to retrieve select catalog sql query", sqlException, sql);
if (msg != null) {
     logger.warning(msg);
}
```

another option maybe better, it can be add something like:

```
logger.log(Level.SEVERE, selectCatalogsSql);
throw sqlException;
```